app/views/protocols/index/_protocol_preview_modal_body.html.erb

```diff
@@ -0,0 +1,179 @@
+                  <hr>
+                  <div class="col-xs-12">
+                    <strong><%= t("protocols.steps.tables") %></strong>
+                    <% step.tables.each do |table| %>
+                      <div data-role="hot-table" class="hot-table">
+                        <%= hidden_field(table, :contents, value: table.contents_utf_8, class: "hot-contents") %>
+                        <div data-role="step-hot-table" class="step-result-hot-table"></div>
+                      </div>
+                    <% end %>
+                  </div>
+                <% end %>
+
+                <% assets = ordered_assets(step) %>
+                <% unless assets.blank? then %>
+                  <hr>
+                  <div class="col-xs-12">
+                    <strong><%= t("protocols.steps.files") %></strong>
+                    <ul class="list-unstyled">
+                      <% assets.each do |asset| %>
+                        <li>
+                          <%= image_tag preview_asset_path(asset) if asset.is_image? %>
+                          <p><%= truncate(asset.file_file_name,
+                                          length: Constants::FILENAME_TRUNCATION_LENGTH) %></p>
+                        </li>
+                      <% end %>
+                    </ul>
+                  </div>
+                <% end %>
+
+                <% unless step.checklists.blank? then %>
+                  <hr>
+                  <div class="col-xs-12">
+                    <% step.checklists.each do |checklist| %>
+                      <strong><%= checklist.name %></strong>
+                      <% if checklist.checklist_items.empty? %>
+                        </br>
+                        <%= t("protocols.steps.empty_checklist") %>
+                        </br>
+                      <% else %>
+                        <% ordered_checklist_items(checklist).each do |checklist_item| %>
+                          <div class="checkbox">
+                            <label>
+                              <% if protocol.in_module? %>
+                                <input type="checkbox" value="" <%= "checked" if checklist_item.checked? %> disabled="disabled"/>
+                              <% else %>
+                                <input type="checkbox" value="" disabled="disabled" />
+                              <% end %>
+                              <%= checklist_item.text %>
+                            </label>
+                          </div>
+                        <% end %>
+                      <% end %>
+                    <% end %>
+                  </div>
+                <% end %>
+
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    <% end %>
+  </div>
+
+  <% if protocol.steps.count == 0 %>
+    <div data-role="no-steps-text">
+      <em><%= t("protocols.steps.no_steps") %></em>
+    </div>
+  <% end %>
+</div>
+
+<%= javascript_include_tag "protocols/steps" %>
```

Comment:
All those CSS & JS includes can probably be thrown out of the modal body (since you already include them on `index.html.erb` page).

Display of handsontables stops working if you remove those lines here, but you should research why is that. I believe they should be removed here (you don't want duplicate includes of CSS & JS).